core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MakeAsyncHttpRequestStage.java

```diff
@@ -173,6 +175,19 @@ public void onError(Throwable error) {
         return responseFuture;
     }
 
+    private CompletableFuture<Void> doExecuteHttpRequest(RequestExecutionContext context, AsyncExecuteRequest executeRequest) {
+        MetricCollector metricCollector = context.metricCollector();
+        long callStart = System.nanoTime();
+        CompletableFuture<Void> httpClientFuture = sdkAsyncHttpClient.execute(executeRequest);
+
+        // Offload the metrics reporting from this stage onto the future completion executor
+        httpClientFuture.whenCompleteAsync((r, t) -> {
+            long duration = System.nanoTime() - callStart;
+            metricCollector.reportMetric(CoreMetric.HTTP_REQUEST_ROUND_TRIP_TIME, Duration.ofNanos(duration));
+        }, futureCompletionExecutor);
+        return httpClientFuture;
```

Comment:
does this account for errors that result in no HTTP request being sent?

Reply:
It seems there's no way to tell if the cause of the error is http request not being sent or http response not being received. I can update it to not report `HTTP_REQUEST_ROUND_TRIP_TIME` if there's an error, WDUT?

Reply:
Yeah I think that's more reasonable. Having this metric in there if the request didn't get fully sent or we didn't receive a response might be misleading.